### PR TITLE
[Feature Store] Add format to `list_feature_sets`

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
Implement and use FeatureSetFormat.
Improve performance with a large number of feature sets by not sending the status dictionary, 
except for the state.

[ML-7866](https://iguazio.atlassian.net/browse/ML-7866)

[ML-7866]: https://iguazio.atlassian.net/browse/ML-7866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/api/db/sqldb/helpers.py [hunk 1]
```diff
@@ -99,10 +99,10 @@ def __init__(self, function):
         self._function = function
         self._cache = {}
 
-    def memoize(self, *args):
+    def memoize(self, *args, **kwargs):
```
Comment (tomerm-iguazio) [inline server/api/db/sqldb/helpers.py:102]: add comment: kwargs is not used in memoize.

File: server/api/db/sqldb/helpers.py [hunk 2]
```diff
@@ -99,10 +99,11 @@ def __init__(self, function):
...
+        # kwargs are not been memoized
```
Comment (gtopper) [inline server/api/db/sqldb/helpers.py]:
```suggestion
        # kwargs are not included in the memoization key
```

File: tests/rundb/test_httpdb.py [hunk 3]
```diff
@@ -593,9 +640,24 @@ def test_feature_sets(create_server):
         partition_order="desc",
     )
     assert len(feature_sets) == count, "bad list results - wrong number of members"
+    assert all([feature_set.status.stats for feature_set in feature_sets])
+    assert all([feature_set.status.preview for feature_set in feature_sets])
 
     feature_set = db.get_feature_set(name, project)
     assert len(feature_set.spec.features) == 4
+    # test minimal feature_sets list:
```
Comment (gtopper) [inline tests/rundb/test_httpdb.py]: It's better for comments to be written in English and not pseudo-code, but I don't think this is even a minimal feature set list? Minimal would be of size zero.
```suggestion
    # test minimal feature set list
```

File: tests/rundb/test_httpdb.py [hunk 4]
```diff
@@ -593,9 +640,24 @@ def test_feature_sets(create_server):
...
+    feature_sets = db.list_feature_sets(
+        project=project,
+        partition_by="name",
+        rows_per_partition=1,
+        partition_sort_by="updated",
+        partition_order="desc",
+        format_=mlrun.common.formatters.FeatureSetFormat.minimal,
+    )
+    assert len(feature_sets) == count, "bad list results - wrong number of members"
```
Comment (gtopper) [inline tests/rundb/test_httpdb.py]: The condition is self-explanatory .
```suggestion
    assert len(feature_sets) == count
```

File: tests/rundb/test_httpdb.py [hunk 5]
```diff
@@ -592,10 +639,25 @@ def test_feature_sets(create_server):
...
         partition_order="desc",
     )
-    assert len(feature_sets) == count, "bad list results - wrong number of members"
+    assert len(feature_sets) == count
+    assert all([feature_set.status.stats for feature_set in feature_sets])
+    assert all([feature_set.status.preview for feature_set in feature_sets])
 
     feature_set = db.get_feature_set(name, project)
     assert len(feature_set.spec.features) == 4
+    # test minimal feature set list
```
Comment (gtopper) [inline tests/rundb/test_httpdb.py]: minimal format, not minimal list

**Review by gtopper — COMMENTED** [review]
Looks good in terms of code, but @tomerm-iguazio is still investigating what's slowing this API down so much, because it's still slower than would be expected even after this change.

File: mlrun/db/httpdb.py [hunk 6]
```diff
@@ -2262,6 +2265,10 @@ def list_feature_sets(
         :param partition_sort_by: What field to sort the results by, within each partition defined by `partition_by`.
             Currently the only allowed value are `created` and `updated`.
         :param partition_order: Order of sorting within partitions - `asc` or `desc`. Default is `desc`.
+                :param format_: Format of the results. Possible values are:
```
Comment (gtopper) [inline mlrun/db/httpdb.py]:
```suggestion
        :param format_: Format of the results. Possible values are:
```

File: mlrun/db/httpdb.py [hunk 7]
```diff
@@ -2262,6 +2265,10 @@ def list_feature_sets(
...
+            - ``minimal`` - Return minimal project objects (minimization happens in the BE), not including
+             stats and preview for each feature set.
+            - ``full``  - Return full project objects.
```
Comment (gtopper) [inline mlrun/db/httpdb.py]:
```suggestion
            - ``full`` - Return full feature set objects.
```

File: mlrun/db/httpdb.py [hunk 8]
```diff
@@ -2262,6 +2265,10 @@ def list_feature_sets(
         :param partition_sort_by: What field to sort the results by, within each partition defined by `partition_by`.
             Currently the only allowed value are `created` and `updated`.
         :param partition_order: Order of sorting within partitions - `asc` or `desc`. Default is `desc`.
+                :param format_: Format of the results. Possible values are:
+            - ``minimal`` - Return minimal project objects (minimization happens in the BE), not including
```
Comment (gtopper) [inline mlrun/db/httpdb.py]:
```suggestion
            - ``minimal`` - Return minimal feature set objects, not including
```

**Review by gtopper — COMMENTED** [review]
Looks good otherwise, but as we discussed, there is room for further improvement by not fetching all this unnecessary data from the database to begin with.

If we fix that, we actually won't need all the filtering in the code.

File: tests/rundb/test_httpdb.py [hunk 9]
```diff
@@ -592,10 +639,25 @@ def test_feature_sets(create_server):
...
         partition_order="desc",
     )
-    assert len(feature_sets) == count, "bad list results - wrong number of members"
+    assert len(feature_sets) == count
+    assert all([feature_set.status.stats for feature_set in feature_sets])
+    assert all([feature_set.status.preview for feature_set in feature_sets])
 
     feature_set = db.get_feature_set(name, project)
     assert len(feature_set.spec.features) == 4
+    # test minimal feature set format
```
Comment (gtopper) [inline tests/rundb/test_httpdb.py:649]:
```suggestion

    # test minimal feature set format
```

**Review by gtopper — APPROVED** [review]
We can move ahead with filtering for the time being. Hopefully we'll be able to drop the columns from the query itself in a later PR though.

**Review by gtopper — APPROVED** [review]

**Pull request merged**